js/packages/mobile-wallet-adapter-protocol/android/src/main/java/com/solanamobile/mobilewalletadapter/reactnative/SolanaMobileWalletAdapterModule.kt

```diff
@@ -55,8 +78,14 @@ class SolanaMobileWalletAdapterModule(reactContext: ReactApplicationContext) :
                 localAssociation.port,
                 localAssociation.session
             )
-            currentActivity?.startActivityForResult(intent, 0)
-                ?: throw NullPointerException("Could not find a current activity from which to launch a local association")
+            launchIntent(intent, REQUEST_LOCAL_ASSOCIATION) { resultCode, _ ->
+                if (resultCode == Activity.RESULT_CANCELED) {
+                    Log.d(name, "Local association cancelled by user, ending session")
+                    promise.reject("Session not established: Local association cancelled by user", 
```

Comment:
This if branch makes sense with a rejection for the promise. But why don't we also have an if branch like:

```
else { // success case
   promise.resolve(true)
}
```

Actually, why doesn't `line 92` with `promise.resolve(true)` lead to a race condition with the completion callback from `launchIntent`? (I'm making the assumption that the completion callback is async)

Reply:
> But why don't we also have an if branch like:

Because we don't care about else case, we only care if the result is cancelled. there is nothing else for us to do in any other case. 

> why doesn't line 92 with `promise.resolve(true)` lead to a race condition with the completion callback from `launchIntent`

Because the `promise.resolve(true)` on line 92 will _never_ be reached if the user cancels the intent. If the wallet never receives the intent, the websocket server will never exist.